apps/coordinator/src/checkpointer.ts

```
@@ -578,30 +558,77 @@ export class Checkpointer {
+
+      this.#logger.error("Unhandled checkpoint error", { options, error });
+
+      return { success: false, reason: "ERROR" };
+    } finally {
+      await cleanup();
+
+      if (controller.signal.aborted) {
+        this.#logger.error("Checkpoint canceled: Cleanup", { options });
+
+        // Overrides any prior return value
         return { success: false, reason: "CANCELED" };
```

Biome

[error] Unsafe usage of 'return'.

'return' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)